tests/basic/labels.rs

```diff
@@ -0,0 +1,337 @@
+use shuttle::{
+    check_dfs, check_random,
+    current::{get_label_for_task, me, set_label_for_task, set_name_for_task, ChildLabelFn, TaskName},
+    future, thread,
+};
+use std::collections::HashSet;
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+use test_log::test;
+use tracing::field::{Field, Visit};
+use tracing::span::{Attributes, Record};
+use tracing::{Event, Id, Metadata, Subscriber};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct Ident(usize);
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct Parent(usize);
+
+// This test does the following: spawns a tree of tasks with integer ids
+// Each task has labels `Parent(i)` and `Ident(j)` where i is the parent's id, and j is the task's id.
+// Here is the tree of tasks spawned:
+//                    (1)
+//                /        \
+//             (2)          (3)
+//            /   \        /   \
+//         (4)    (5)    (6)   (7)
+//
+// Each leaf task increments a global AtomicUsize (initially 0) by 1
+// Each task returns its (Parent, Ident) as a usize pair when it completes.
+// The root waits for all tasks to complete, and collects the returned usize pairs. It then checks
+// - that the final AtomicUsize value is 4
+// - that each (parent, child) pair in the tree above is reported exactly once
+//
+// This test checks the following properties
+// - tasks inherit the Labels of their parent
+// - changing child Labels doesn't affect parent Labels
+// - child Labels are independent of each other
+async fn spawn_tasks(counter: Arc<AtomicUsize>) -> HashSet<(usize, usize)> {
+    set_label_for_task(me(), Ident(1));
+    #[allow(clippy::type_complexity)]
+    let handles: Vec<future::JoinHandle<(usize, usize, Vec<future::JoinHandle<(usize, usize)>>)>> = (0..2)
```

Comment:
This resolves with `Vec<_>`

Reply:
https://github.com/awslabs/shuttle/commit/496b60e92566a8ae5ba265ca94a1b0e5d2b7f4a2

Reply:
Nice catch; thanks